presto-main/src/main/java/com/facebook/presto/cost/PlanNodeCostEstimate.java

```diff
@@ -30,6 +30,10 @@
     private static final PlanNodeCostEstimate UNKNOWN = new PlanNodeCostEstimate(NaN, NaN, NaN);
     private static final PlanNodeCostEstimate ZERO = new PlanNodeCostEstimate(0, 0, 0);
 
+    private final double cpuCost;
+    private final double memoryCost;
+    private final double networkCost;
```

Comment:
i prefer to keep static factory methods _above_ the instance fields (and below class constants)

Reply:
Everywhere else in the code the static factory methods are below the fields. 

(`Symbol`, `ExpressionInterpreter`, `TupleDomain`, `Domain`, `MetadataManager`, ...)

Some clases define static factor methods after constructor, but i don't think it is right. As we try to define methods in the order of they appearance. So if method A uses method B, method A comes first and the method B comes second. Similar with static factory methods. Since factory methods use constructor, they should come before constructor.

Reply:
It would be nice to have a build check for that to not to go back and forth